Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm

```diff
@@ -205,17 +194,21 @@ static void clearNoSenderNotifications(mach_port_t port)
         m_isConnected = true;
 
         // Send the initialize message, which contains a send right for the server to use.
-        auto encoder = makeUniqueRef<Encoder>(MessageName::InitializeConnection, 0);
-
         mach_port_insert_right(mach_task_self(), m_receivePort, m_receivePort, MACH_MSG_TYPE_MAKE_SEND);
-        encoder.get() << MachSendRight::adopt(m_receivePort);
+        auto serverSendRight = MachSendRight::adopt(m_receivePort);
```

Comment:
This jump from native mach calls, into using our wrapper took a bit to wrap my head around (we're adopting just the newly allocated send right, but the port still has the existing receive right attached, so `m_receivePort` isn't invalid).

That's not new with your changes,  but it'd kinda be nice if we had `MachReceiveRight` which could then give us a newly allocated `MachSendRight`. One day :)

Reply:
If I ever understand the exact semantics of the port and port right namespaces, I'll implement it.